cinema/tests/test_movie_api.py

```diff
@@ -157,3 +156,191 @@
+        movie_with_genre_2 = sample_movie(title="test2",
+                                          description="test2",
+                                          duration=90)
+        genre_1 = sample_genre(name="test1")
+        genre_2 = sample_genre(name="test2")
+
+        movie_with_genre_1.genres.add(genre_1)
+        movie_with_genre_2.genres.add(genre_2)
+
+        res = self.client.get(
+            MOVIE_URL,
+            {"genres": f"{genre_1.id},{genre_2.id}"},
+        )
+
+        serializer_without_genres = MovieListSerializer(movie_without_genre)
+        serializer_movie_genre_1 = MovieListSerializer(movie_with_genre_1)
+        serializer_movie_genre_2 = MovieListSerializer(movie_with_genre_2)
+
+        self.assertIn(serializer_movie_genre_1.data, res.data)
+        self.assertIn(serializer_movie_genre_2.data, res.data)
+        self.assertNotIn(serializer_without_genres.data, res.data)
+
+    def test_filter_movies_by_actor(self):
+        movie_without_actor = sample_movie()
+        movie_with_actor_1 = sample_movie(title="test1",
+                                          description="test1",
+                                          duration=90)
+        movie_with_actor_2 = sample_movie(title="test2",
+                                          description="test2",
+                                          duration=90)
+        actor_1 = sample_actor(first_name="test1", last_name="test1")
+        actor_2 = sample_actor(first_name="test2", last_name="test2")
+
+        movie_with_actor_1.actors.add(actor_1)
+        movie_with_actor_2.actors.add(actor_2)
+
+        res = self.client.get(
+            MOVIE_URL,
+            {"actors": f"{actor_1.id},{actor_2.id}"},
+        )
+
+        serializer_without_actor = MovieListSerializer(movie_without_actor)
+        serializer_movie_actor_1 = MovieListSerializer(movie_with_actor_1)
+        serializer_movie_actor_2 = MovieListSerializer(movie_with_actor_2)
+
+        self.assertIn(serializer_movie_actor_1.data, res.data)
+        self.assertIn(serializer_movie_actor_2.data, res.data)
+        self.assertNotIn(serializer_without_actor.data, res.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.genres.add(sample_genre(name="test1"))
+        movie.actors.add(sample_actor(first_name="test1", last_name="test1"))
+
+        url = detail_url(movie.id)
+
+        res = self.client.get(url)
+
+        serializer = MovieDetailSerializer(movie)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "test",
+            "description": "test",
+            "duration": 90,
+        }
+
+        res = self.client.post(MOVIE_URL, payload)
+
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieTests(TestCase):
```

Comment:
The `AdminMovieTests` class tests the API functionality for admin users, including creating movies with genres and actors. This is crucial for verifying that admin-specific operations are correctly implemented.